cluster/manager_delete.go

```diff
@@ -61,7 +91,7 @@ func (m *Manager) deleteCluster(ctx context.Context, cluster CommonCluster, forc
 
 	if !(force && c == nil) {
 		// delete deployments
-		err = helm.DeleteAllDeployment(c)
+		err = deleteAllResource(c, logger)
```

Comment:
Add retry logic to handle intermittent issues like unauthorized in case of EKS

Reply:
Retry logic should probably part of the function itself.

Also, with a bit of code compressing you can make retrying easier:

```go
type resourceDeleter interface {
	Delete(name string, options metav1.DeleteOptions)
}

services := []resourceDeleter{
	client.CoreV1().Services(""),
	client.AppsV1().Deployments(""),
	// ...
}

options := metav1.NewDeleteOptions(0)

for service := range services {
	err := service.Delete(options)
	if err != nil {
		// retry?
	}
}
```

Not sure if this is ultimately better.